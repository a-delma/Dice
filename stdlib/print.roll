import "../stdlib/math.roll"; 

[] -> Void newline;
[] -> Void dot;
[Int] -> Void printDigit;
[Int] -> Void printInt;
[Float] -> Float mantissa;
[Float] -> Void printMantissa;
[Float] -> Void printFloat;

//how many decimal points to print for floats
Int floatPrecision;
floatPrecision = 4;

newline = lambda()->Void{ putChar(10); };
dot = lambda()->Void{ putChar(46); };

//Only works on single digits, for internal use
//  to use with printInt
printDigit = lambda(Int d)->Void{
  putChar(48 + d);
};

printInt = lambda(Int n)->Void{
  if(n < 10){
    printDigit(n);
  } else {
    printDigit(mod(n, 10));
    self(n/10); 
  }
};

mantissa = lambda()->Float{
  //stub
  return 9.234;
}

//should take a float less than 1.
printMantissa = lambda(Float f, Int remaining)->Void{
  if(remaining < 1){
    //just return 
  } else {
    printDigit(floatToInt(f * 10));
    //self( , remaining - 1);
  }
};

printFloat = lambda(Float f)->Void{
  Int theInt;
  theInt = floatToInt(f);
  printInt(theInt);
  dot();
};
