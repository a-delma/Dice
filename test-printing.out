H
H.
1234
0.