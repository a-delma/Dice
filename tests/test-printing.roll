import "../stdlib/print.roll";
//import "../stdlib/math.roll";

putChar(72);
newline();
putChar(72);
dot();
newline();
printInt(1234);