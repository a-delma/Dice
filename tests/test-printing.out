H
H.
1234